UnitTests/BraintreeThreeDSecureTests/BTThreeDSecureClient_Tests.swift

```diff
@@ -253,8 +253,29 @@ class BTThreeDSecureClient_Tests: XCTestCase {
         waitForExpectations(timeout: 2)
         XCTAssertTrue(mockAPIClient.postedAnalyticsEvents.contains(BTThreeDSecureAnalytics.verifyFailed))
     }
+
+    func testStartPaymentFlow_whenAmountIsNil_throwsError() {
+        mockAPIClient.cannedConfigurationResponseBody = BTJSON(value: [
+            "threeDSecure": ["cardinalAuthenticationJWT": "FAKE_JWT"],
+            "assetsUrl": "http://assets.example.com"
+        ] as [String: Any])
+
+        let request = BTThreeDSecureRequest()
+        request.amount = nil
+
+        let expectation = expectation(description: "Callback envoked")
+
+        client.startPaymentFlow(request) { result, error in
+            XCTAssertNil(result)
+            XCTAssertEqual(error?.localizedDescription, "BTThreeDSecureRequest amount can not be nil or NaN.")
+            expectation.fulfill()
+        }
+
+        waitForExpectations(timeout: 2)
+        XCTAssertTrue(mockAPIClient.postedAnalyticsEvents.contains(BTThreeDSecureAnalytics.verifyFailed))
+    }
     
-    func testStartPayment_returnsError_whenAmountIsMissing() {
+    func testStartPayment_whenNoBodyReturned_returnsAnError() {
```

Comment:
This was actually testing `noBodyReturned` and not missing amount, updated the test name and added the test `testStartPaymentFlow_whenAmountIsNil_throwsError` to test what this test originally said it was testing.